### PR TITLE
Skins mode for NanoVNA Saver

### PR DESCRIPTION
**Pull request by cinosh07**
Hi,

i've added a dark mode for the ui and color mode for value fields.
Those modes can be configured in the display setup dialog.

qtmodern library cannot be installed from python package repository cause it's outdated.
I've change pipfile to get this library from github.
You can install it manually with this command: 

`pip3 install git+https://github.com/gmarull/qtmodern.git`

colored style can be edited in the light-colored.css and dark-colored.css files.

template
template2

### CODE REVIEW

**Comment by mihtjel** [comment]
Hello Carl,
thank you very much! This looks really good! Well, other than the readings being a bit iffy? ;-)

I have a couple of issues, however:
The minor one is that your pull request includes both a number of cache files, as well as two screenshots that don't seem to play a role in the code. These should probably go :-)
The more major one is the one of using qtmodern. As you write, it's not available in the regular pip repositories, and I am not very fond of the idea of pulling the code direct from GitHub for it. While it does have commits within the last couple of months, I'm not entirely certain it's a maintained project?

I am also a little irked by changing away from the user interface standard that the users themselves have selected. Modern OSes offer some amount of "theming" of UI elements, and overriding that is not something that should be done lightly. However much I like the look of qtmodern better. ;-)

If it were possible to extract the qtmodern to be an optional addition, imported within an "if" clause, for example, so the user could at launch select to launch the "classic" or "modern" versions, this would also allow me to build Windows versions of both setups.  If the user base were to shift entirely, this would be a good indication for a full switch at that point.

Regardless of what solution we end up with, I think I will take your additions regarding CSS classing of the individual labels on board. Thank you very much for those!

Thank you for contributing!
Rune / 5Q5R

File: NanoVNASaver/SweepWorker.py [hunk 1]
```diff
@@ -271,7 +271,9 @@ def truncate(self, values: List[List[tuple]], count):
     def readSegment(self, start, stop):
         logger.debug("Setting sweep range to %d to %d", start, stop)
         self.app.setSweep(start, stop)
-        sleep(1)    # TODO This long delay seems to fix the weird data transitions we were seeing by getting partial
+
+        sleep(0.3)    # TODO This is fixed in 0.2.3 firmware
+        #sleep(1)    # TODO This long delay seems to fix the weird data transitions we were seeing by getting partial
```
Comment (mihtjel) [inline NanoVNASaver/SweepWorker.py]: This is fixed in a different manner in the development branch.

**Comment by cinosh07** [comment]
Hi Rune, 

That's a wonderfull job that you did on this.
I've removed the need for qtmodern.
Default display is the actual default UI.

Added the concept of skins to the UI.

Added Skins controls in the display settings dialog. Skin Mode is mutually exclusive with dark mode custom color mode. So when Skins is active, dark mode and custom color mode are disabled. Skin changes is not working yet. Rune i'll answer your comments as soon as skins changes is fixed. So we can chat further more on future.

Still have work to for the skin change function. It's not working yet.

For the e-mail you sent, i'll answer later today and will find solutions. 

That's becoming a very great app.

**Comment by mihtjel** [comment]
You've done great work, and it's a pleasure to have someone else working on the software as well :-)

**Comment by cinosh07** [comment]
Hi Rune,

Thanks for the good words.

I think I've met all your requirements.

Skins feature completed and ready to review and test.

Improved UI Quality in the skins.

Skins css files located in the skins folder.

Added author and contributors to the about dialog.

When in debug mode, css files are get from os path else from sys._MEIPASS as you asked.

Known issue: The app need to be restarted when return to default UI when you change from a skinned UI

73 VA2SAJ

UI-1

UI-2

**Comment by mihtjel** [comment]
Hi Carl,
it looks great! I'll put some in-line comments in some of the changes, but there's one general one: I haven't been idling while you were doing this, so there's a Development branch of the code which is different from the master branch you developed this against. This is currently for the 0.1.2 release which is due Soon<tm>. So we will need to get the changes here merged with the changes there.

73,
Rune

File: NanoVNASaver/Calibration.py [hunk 2]
```diff
@@ -37,8 +37,7 @@ def __init__(self, app):
         from .NanoVNASaver import NanoVNASaver
 
         self.app: NanoVNASaver = app
-
-        self.setMinimumSize(450, 600)
+        self.setMinimumSize(450, 800)
```
Comment (mihtjel) [inline NanoVNASaver/Calibration.py]: Can't have minimum height be more than 700-ish. I would prefer not increasing it at all unless strictly needed.

File: NanoVNASaver/NanoVNASaver.py [hunk 3]
```diff
@@ -36,6 +37,10 @@
 from .Touchstone import Touchstone
 from .Analysis import Analysis, LowPassAnalysis, HighPassAnalysis, BandPassAnalysis
 from .about import version as ver
+from PyQt5.QtWidgets import QApplication, QPushButton, QDesktopWidget
```
Comment (mihtjel) [inline NanoVNASaver/NanoVNASaver.py]: I generally refer to the widgets as QtWidgets.QPushButton, for example - so these should probably be removed.

File: NanoVNASaver/NanoVNASaver.py [hunk 4]
```diff
@@ -13,6 +13,7 @@
 #
 #  You should have received a copy of the GNU General Public License
 #  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+# import os
```
Comment (mihtjel) [inline NanoVNASaver/NanoVNASaver.py]: Commented out, so could be removed?

File: NanoVNASaver/NanoVNASaver.py [hunk 5]
```diff
@@ -475,7 +489,8 @@ def __init__(self):
                                                                       "\n\n\N{COPYRIGHT SIGN} Copyright 2019 Rune B. Broberg\n" +
                                                                       "This program comes with ABSOLUTELY NO WARRANTY\n" +
                                                                       "This program is licensed under the GNU General Public License version 3\n\n" +
-                                                                      "See https://mihtjel.github.io/nanovna-saver/ for further details"))
+                                                                      "See https://mihtjel.github.io/nanovna-saver/ for further details\n\n" +
+                                                                      "AUTHOR:\nRune B. Broberg - 5Q5R\n\nCONTRIBUTORS:\nOhan Smit, Neilkatin, Carl Tremblay - VA2SAJ"))
```
Comment (mihtjel) [inline NanoVNASaver/NanoVNASaver.py]: This is good, but needs to be ported to the new About window :-)

File: NanoVNASaver/skins_utils.py [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+    if debug == True:
+        return join(dirname(abspath(__file__)), 'skins/light-monochrome.css')
+    else:
+        return sys._MEIPASS + '/skins/light-monochrome.css'
+
+def LIGHT_SKIN_COLORED():
+    if debug == True:
+        return join(dirname(abspath(__file__)), 'skins/light-colored.css')
+    else:
+        return sys._MEIPASS + '/skins/light-colored.css'
```
Comment (mihtjel) [inline NanoVNASaver/skins_utils.py]: The sys._MEIPASS variable is only used if packaged via pyinstaller. See the corresponding code in NanoVNASaver.py loading the icon for how to determine if the code is running within such an environment. "Normally" it should just load from a subdirectory. I think it's best to let that be a subdirectory of the application working directory? But I'm open to being educated on what's best practice.

File: NanoVNASaver/Calibration.py (hunk 2 above)
Comment (cinosh07) [reply NanoVNASaver/Calibration.py]: To do so I will have to put css class to each component of the calibration window. Now it use the global values for those type of components.

File: NanoVNASaver/skins_utils.py (hunk 6 above)
Comment (cinosh07) [reply NanoVNASaver/skins_utils.py]: Done ready for the next commit

**Comment by cinosh07** [comment]
Closing this pull request and will redo another one for dev branch

File: NanoVNASaver/NanoVNASaver.py (hunk 3 above)
Comment (cinosh07) [reply NanoVNASaver/NanoVNASaver.py]: done

File: NanoVNASaver/Calibration.py (hunk 2 above)
Comment (cinosh07) [reply NanoVNASaver/Calibration.py]: done

File: NanoVNASaver/NanoVNASaver.py (hunk 4 above)
Comment (cinosh07) [reply NanoVNASaver/NanoVNASaver.py]: done

File: NanoVNASaver/NanoVNASaver.py (hunk 5 above)
Comment (cinosh07) [reply NanoVNASaver/NanoVNASaver.py]: done

**Pull request closed without merging**